File: 프로그래머스/lv1/이재표/소수 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int n = 0;
        for (int i = 0; i <= 1000; i++) {
            n += i;
        }
        
        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;

        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        int answer = 0;
        for(int i=0;i<nums.length;i++){
            for(int j=i+1;j<nums.length;j++){
                for(int k=j+1;k<nums.length;k++){
                    int sum=nums[i]+nums[j]+nums[k];
                    if(isPrime[sum]){
                        answer++;
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv1/김미래/소수 만들기.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int solution(vector<int> nums) {
    int answer = 0;
    vector<bool> temp(nums.size(), false);
    fill(temp.end()-3, temp.end(), true);
    do{
        int result = 0;
        for(int i=0; i<nums.size(); i++){
            if(temp[i]==true){
                result +=nums[i];
            }
        }
        
        for(int j=2; j<result; j++){
            if(result % j == 0){
                break;
            }
            if(j == result -1){
                answer++;
            }
        }
    }while(next_permutation(temp.begin(), temp.end()));

    return answer;
}
